src/ImageData/FitsLoader.h

```diff
@@ -207,6 +213,71 @@ void FitsLoader::RemoveHistoryBeam(unsigned int hdu_num) {
     }
 }
 
+bool FitsLoader::Is64BitBeamTable(const std::string& filename) {
+    fitsfile* fptr;
+    int status(0);
+    fits_open_file(&fptr, filename.c_str(), 0, &status);
+
+    if (status) {
+        spdlog::error("Error opening FITS file.");
+        return false;
+    }
+
+    // Open binary table extension with name BEAMS
+    int hdutype(BINARY_TBL), extver(0);
+    std::string extname("BEAMS");
+    status = 0;
+    fits_movnam_hdu(fptr, hdutype, extname.data(), extver, &status);
+
+    if (status) {
+        status = 0;
+        fits_close_file(fptr, &status);
+        spdlog::info("Could not find BEAMS table.");
+        return false;
+    }
+
+    // Get BEAMS columns data type
+    int casesen(CASEINSEN), colnum(0);
+#if defined(__APPLE__)
+    char ttype, tform;
+#else
+    char* ttype = (char*)malloc(sizeof(char*));
+    char* tform = (char*)malloc(sizeof(char*));
+#endif
+    char tunit, nulstr, tdisp;
+    long tbcol;
+    double scale, zero;
+
+    auto close_file = [&]() {
+        fits_close_file(fptr, &status);
+#if defined(__APPLE__)
+#else
+        free(ttype);
+        free(tform);
+#endif
+    };
+
+    std::vector<std::string> keys = {"BMAJ", "BMIN", "BPA", "CHAN", "POL"};
+    for (auto& key : keys) {
+        status = 0;
+        fits_get_colnum(fptr, casesen, key.data(), &colnum, &status);
+#if defined(__APPLE__)
+        fits_get_acolparms(fptr, colnum, &ttype, &tbcol, &tunit, &tform, &scale, &zero, &nulstr, &tdisp, &status);
```

Comment:
Would fits_get_coltype() be easier, then check if typecode is TFLOAT or TDOUBLE?  Rather than malloc/free char* arguments.

int fits_get_coltype / ffgtcl
      (fitsfile *fptr, int colnum, > int *typecode, long *repeat,
       long *width, int *status)